lib/credentials/token_file_web_identity_credentials.js

```diff
@@ -84,11 +84,24 @@ AWS.TokenFileWebIdentityCredentials = AWS.util.inherit(AWS.Credentials, {
       );
     }
 
-    return {
+    var paramsArray = [];
+
+    while (!profile['web_identity_token_file'] && profile['source_profile']) {
+      paramsArray.unshift({
+        roleArn: profile['role_arn'],
+        roleSessionName: profile['role_session_name']
```

Comment:
Is there any other parameters you need to send with `assumerole` call? From what I can find [SharedIniCredentials](https://github.com/aws/aws-sdk-js/blob/88b635be22df3207ab7971d9b3192b5523a0a8b8/lib/credentials/shared_ini_file_credentials.js#L229-L231) can also take `ExternalId`.

Reply:
Nope, only RoleArn and RoleSessionName are necessary for this use case https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/STS.html#assumeRole-property

Reply:
Confirmed internally that only RoleArn and RoleSessionName are necessary for assumeRole call